pac4j-kerberos/src/main/java/org/pac4j/kerberos/credentials/extractor/KerberosExtractor.java

```diff
@@ -0,0 +1,53 @@
+package org.pac4j.kerberos.credentials.extractor;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import org.pac4j.kerberos.credentials.KerberosCredentials;
+import org.pac4j.core.context.HttpConstants;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.credentials.extractor.CredentialsExtractor;
+import org.pac4j.core.exception.CredentialsException;
+import org.pac4j.core.exception.DeferredHttpAction;
+import org.pac4j.core.exception.DeferredHttpActionCallback;
+
+/**
+ * To extract Kerberos headers.
+ *
+ * @author Garry Boyce
+ * @since 1.9.1
+ */
+public class KerberosExtractor implements CredentialsExtractor<KerberosCredentials> {
+    private final String clientName;
+
+    public KerberosExtractor(final String clientName) {
+        this.clientName = clientName;
+    }
+
+    @Override
+    public KerberosCredentials extract(WebContext context) {
+        final String header = context.getRequestHeader(HttpConstants.AUTHORIZATION_HEADER);
+        if (header == null) {
+            throw DeferredHttpAction.deferredHttpAction("Kerberos Header not found", new DeferredHttpActionCallback() {

```

Comment:
Notice that the new spring-security-pac4j v2 library has strongly changed and the entry point does no longer matter. You now have the regular `SecurityFilter` and `CallbackFilter`.

You can use both clients (`DirectBasicAuth` and `IndirectBasicAuthClient`)  in the `SecurityFilter` for advanced use cases.

Examples:

`clients = "DirectBasicAuth"` : a basic auth is expected and if you have none, a 401 error page is returned
`clients = "IndirectBasicAuthClient"` :  the user is expected to be authenticated, if not, a redirection happens to the callback url where the indirect client asks for basic auth credentials
`clients = "IndirectBasicAuthClient,DirectBasicAuthClient,HeaderClient"` : a basic auth is expected, if none, a header is expected and if none, a redirection happens to the callback url where the indirect client asks for basic auth credentials: isn't it the use case you try to address?